packages/runtime/container-runtime/src/metadata.ts

```diff
@@ -3,25 +3,21 @@
  * Licensed under the MIT License.
  */
 
-/**
- * Does the metadata object look like batch metadata?
- */
-export function isBatchMetadata(metadata: any): metadata is IBatchMetadata {
-	return typeof metadata?.batch === "boolean";
-}
+import type { BatchId } from "./opLifecycle/index.js";
 
-/**
- * Cast the given metadata object to IBatchMetadata if it is so, otherwise yield undefined
- */
+/** Syntactic sugar for casting */
 export function asBatchMetadata(metadata: unknown): IBatchMetadata | undefined {
-	return isBatchMetadata(metadata) ? metadata : undefined;
+	return metadata as IBatchMetadata | undefined;
```

Comment:
For some reason, I favored the older solution as it seemed a little safer, but I don't know if this is important.

Reply:
Yeah I had just added that type guard, but on second though didn't think the runtime checks were that interesting.  The only risk is if two different parts of code used the same key `batch` or `batchId` and stored data with a different type and different semantics.  That's not going to happen though...  So let's not waste time with the runtime checks.

Since all `IBatchMetadata` props are optional, and this thing indicates it could be undefined, callers have to handle those cases.

Maybe it's better to return `Partial<IBatchMetadata> | undefined` to emphasize the uncertainty (basically just want callers to be able to get intellisense on valid property names + expected types)